packages/server/src/open-feature.ts

```diff
@@ -71,22 +73,107 @@ export class OpenFeatureAPI extends OpenFeatureCommonAPI {
     return this;
   }
 
-  setProvider(provider: Provider): this {
-    // ignore no-ops
-    if (this._provider !== provider) {
+  /**
+   * Sets the provider that OpenFeature will use for flag evaluations of clients without a name.
+   * Setting a provider supersedes the current provider used in new and existing clients without a name.
+   *
+   * @param {Provider} provider The provider responsible for flag evaluations.
+   * @returns {GlobalApi} OpenFeature API
+   */
+  setProvider(provider: Provider): this;
+  /**
+   * Sets the provider that OpenFeature will use for flag evaluations of clients with the given name.
+   * Setting a provider supersedes the current provider used in new and existing clients with that name.
+   *
+   * @param {string} clientName The name to identify the client
+   * @param {Provider} provider The provider responsible for flag evaluations.
+   * @returns {GlobalApi} OpenFeature API
+   */
+  setProvider(clientName: string, provider: Provider): this;
+  setProvider(clientOrProvider?: string | Provider, providerOrUndefined?: Provider): this {
+    const client = stringOrUndefined(clientOrProvider);
+    const provider = objectOrUndefined<Provider>(clientOrProvider) ?? objectOrUndefined<Provider>(providerOrUndefined);
```

Comment:
We might be able to make more of of the `setProvider` function common when we add `events` and `init/shutdown` features to the server, with some help from generics.

Reply:
Makes sense, I also thought about that and would try to add it in the events PR.
Does that make sense to you?

Reply:
> Makes sense, I also thought about that and would try to add it in the events PR. Does that make sense to you?

Yes, or even after as a pure refactor. I don't mind either way.